src/engine/BMDie.php

```diff
@@ -290,7 +290,13 @@ public function roll($isTriggeredByAttack = FALSE, $isSubdie = FALSE) {
 
         if (!isset($this->value) ||
             ($this->doesReroll && !$this->has_flag('JustPerformedTripAttack'))) {
-            $this->set__value(bm_rand($this->min, $this->max));
+            $hookResultArray = $this->run_hooks(__FUNCTION__, array('die' => $this));
+
+            // if all hook results are FALSE, then roll the die
+            if (empty($hookResultArray) ||
+                (0 == count(array_filter($hookResultArray, function ($value) { return $value !== FALSE; })))) {
+                $this->set__value(bm_rand($this->min, $this->max));
+            }
```

Comment:
The point here is that the FALSE response from the hook is telling the code not to reroll the die *because* the die value has been set within the die hook (which currently can only be triggered by Maximum).